desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py

```diff
@@ -251,24 +251,26 @@ def execute(self, notebook, snippet):
           'database': snippet['database'],
         }
       )
-
-    result = connection.execute(statement)
-
-    logs = [message for message in result.cursor.fetch_logs()] if result.cursor and hasattr(result.cursor, 'fetch_logs') else []
-    cache = {
-      'logs': logs,
-      'connection': connection,
-      'result': result,
-      'meta': [
-        {
-          'name': col[0] if (type(col) is tuple or type(col) is dict) else col.name if hasattr(col, 'name') else col,
-          'type': 'STRING_TYPE',
-          'comment': ''
-        }
-        for col in result.cursor.description
-      ] if result.cursor else []
-    }
-    CONNECTIONS[guid] = cache
+    stmt_list = statement.split(';')
+    for stmt in stmt_list:
+
+      result = connection.execute(stmt)
+
+      logs = [message for message in result.cursor.fetch_logs()] if result.cursor and hasattr(result.cursor, 'fetch_logs') else []
+      cache = {
+        'logs': logs,
+        'connection': connection,
+        'result': result,
+        'meta': [
+          {
+            'name': col[0] if (type(col) is tuple or type(col) is dict) else col.name if hasattr(col, 'name') else col,
+            'type': 'STRING_TYPE',
+            'comment': ''
+          }
+          for col in result.cursor.description
+        ] if result.cursor else []
+      }
+      CONNECTIONS[guid] = cache
```

Comment:
Need the direction on how to do it in a better way. @romainr

Reply:
What is the issue? Let's have 10 UPSERTs for now

Reply:
Oh, I missed it that it was in sql_alchemy.py, we can't do that here and must keep it single statement execution.

Multi statement should already be supported, e.g. `job_handle = job_nb.execute(request, batch=False)` in api3.py, consoliddate with https://github.com/cloudera/hue/blob/master/desktop/libs/indexer/src/indexer/indexers/phoenix_sql.py ?

Reply:
I don't think so it is helpful in executing multiple statements together. And I tried that same thing and is it is not working @romainr

Reply:
As we are doing the same in local_file also https://github.com/cloudera/hue/blob/master/desktop/libs/indexer/src/indexer/indexers/sql.py#L397-L400

Reply:
Ok, up to you but we can't modify/hack sql_alchemy.py for sure.